### DISCUSSION TITLE
Can the ArcylicBlur effect on windows works on self-contained binary?

### DISCUSSION THREAD

**Discussion by hv0905** [AvaloniaUI/Avalonia#10308] [Q&A] [answered]
I've just wrote a modern-like window, using the example code mainly from [the tutorial](https://docs.avaloniaui.net/tutorials/music-store-app/creating-a-modern-looking-window).

I add the following properties into my window root of `MainWindow.xaml` file:
```axaml
        TransparencyLevelHint="AcrylicBlur"
        ExtendClientAreaToDecorationsHint="True"
        Background="Transparent"
```

Then add an Arcylic material:
```axaml
<ExperimentalAcrylicBorder IsHitTestVisible="False">
      <ExperimentalAcrylicBorder.Material>
        <ExperimentalAcrylicMaterial
            BackgroundSource="Digger"
            TintColor="Black"
            TintOpacity="1"
            MaterialOpacity="0.65" />
      </ExperimentalAcrylicBorder.Material>
    </ExperimentalAcrylicBorder>
```

It works fine when I run the app, on both `Debug` and `Release` configuration:  
![image](https://user-images.githubusercontent.com/29349119/218261919-4f85b1a3-f5b6-421e-a4fe-fb8cb1a117b8.png)

However, when I try to publish my app, I met some problem.

When I use the following command to publish a non-self-contained win-x64 app:
```bash
dotnet publish -c Release
```

The Arcylic effect works just fine.

However, when I try to use the following command to publish a self-contained win-x64 app:

```bash
dotnet publish -c Release --self-contained
```

I lost my Arcylic effect:
![image](https://user-images.githubusercontent.com/29349119/218262101-84a9bc97-e007-4705-a368-bea4c53c2420.png)

Is this somehow a bug or a technical limitation? Can I use the arcylic effect on the self-contained win32 binary?

My library version (.Net SDK version is net6.0):
```xml
    <PackageReference Include="Avalonia" Version="11.0.0-preview5" />
    <PackageReference Include="Avalonia.Desktop" Version="11.0.0-preview5" />
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.0-preview5" />
    <PackageReference Include="Avalonia.ReactiveUI" Version="11.0.0-preview5" />
    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0-preview5" />
    <PackageReference Include="MessageBox.Avalonia" Version="2.3.1-prev2" />
    <PackageReference Include="Projektanker.Icons.Avalonia" Version="6.4.0-preview5" />
    <PackageReference Include="Projektanker.Icons.Avalonia.FontAwesome" Version="6.4.0-preview5" />
    <PackageReference Include="XamlNameReferenceGenerator" Version="1.5.1" />
```

**Comment by jp2masa** [comment, marked as answer]
You have to publish for `win7-x64`. This was fixed after preview 5 in #10204.

**Reply by hv0905** [reply]
Thanks!
